frontend/pkg/frontend/frontend.go

```diff
@@ -399,12 +460,28 @@ func (f *Frontend) ArmResourceDelete(writer http.ResponseWriter, request *http.R
 	resourceID := request.URL.Path
 	subscriptionID := request.PathValue(PathSegmentSubscriptionID)
 
-	_, found := f.cache.GetCluster(resourceID)
-	if !found {
-		writer.WriteHeader(http.StatusNotFound)
-		return
+	var doc *database.HCPOpenShiftClusterDocument
+	doc, err = f.dbClient.GetClusterDoc(ctx, resourceID, subscriptionID)
+	if err != nil {
+		if errors.Is(err, database.ErrNotFound) {
```

Comment:
Similar comment as above -- maybe we want to ask Cluster Service to search first.